### DISCUSSION TITLE
Query Projection

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [bpmnServer/bpmn-server#198] [Q&A] [answered]
Hi @ralphhanna,

How to use projection when querying properly?

```js
await bpmnClient.datastore.findInstances({
                query: { id: inputs.instId, 'data.caseId': inputs.caseId},
                projection: { 'data.templates': 1 },
            });

```

As a result, I am only interested in the field `data.templates`, but I get everything? Is projection not supported or am I doing anything wrong?

Thank you.

**Comment by ralphhanna** [comment, marked as answer]
Thanks for reporting this,
Projection is now supported in latest release 2.2.4
Thanks

**Reply by MaheshkumarSundaram** [reply]
Thanks

**Comment by ralphhanna** [comment]
This requires new Release of bpmn-web as well.
